ProcessLib/BoundaryCondition/DirichletBoundaryConditionWithinTimeInterval.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * File:   DirichletBoundaryConditionWithinTimeInterval.cpp
+ *
+ * Created on November 26, 2018, 4:59 PM
+ */
+#include "DirichletBoundaryConditionWithinTimeInterval.h"
+
+#include "BaseLib/ConfigTree.h"
+
+#include "NumLib/DOF/LocalToGlobalIndexMap.h"
+#include "NumLib/IndexValueVector.h"
+#include "NumLib/TimeStepping/TimeInterval.h"
+
+#include "ProcessLib/Parameter/Parameter.h"
+#include "ProcessLib/Utils/ProcessUtils.h"
+
+namespace ProcessLib
+{
+DirichletBoundaryConditionWithinTimeInterval::
+    DirichletBoundaryConditionWithinTimeInterval(
+        double const start_time, double const end_time,
+        Parameter<double> const& parameter, MeshLib::Mesh const& bc_mesh,
+        NumLib::LocalToGlobalIndexMap const& dof_table_bulk,
+        int const variable_id, int const component_id)
+    : DirichletBoundaryCondition(parameter, bc_mesh, dof_table_bulk,
+                                 variable_id, component_id),
+      _time_interval(
+          std::make_unique<NumLib::TimeInterval>(start_time, end_time))
+{
+}
+
+void DirichletBoundaryConditionWithinTimeInterval::getEssentialBCValues(
+    const double t, GlobalVector const& x,
+    NumLib::IndexValueVector<GlobalIndexType>& bc_values) const
+{
+    if (_time_interval->isInThisTimeInterval(t))
+    {
+        return getEssentialBCValuesLocal(t, x, bc_values);
```

Comment:
The return type of the function is `void`...

Reply:
changed.